### PR TITLE
Allow WCSCoordinates to init from only WCS

### LINKED ISSUE #1170: Allow WCSCoordinates to be initialized with only a WCS object
At the moment, WCSCoordinates **requires** a header, but we should change this so that a WCS object alone should be enough.

### PR DESCRIPTION
**Pull request by pllim**
Fix #1170. Supersedes #1171.

### CODE REVIEW

File: glue/core/coordinates.py [hunk 1]
```diff
@@ -199,18 +200,31 @@ class WCSCoordinates(Coordinates):
     standard.  This class does not take into account
     distortions.
 
+    Parameters
+    ----------
+    header
+        FITS header (derived from WCS if not given)
```
Comment (astrofrog) [inline glue/core/coordinates.py]: Can you add the type, i.e:

```
header : :class:`astropy.io.fits.Header`, optional
```

and same for ``wcs``?

File: glue/core/coordinates.py [hunk 2]
```diff
@@ -358,10 +377,16 @@ def _get_ndim(header):
```
Comment (astrofrog) [inline glue/core/coordinates.py]: This should be an astropy WCS object, not just any object

**Comment by pllim** [comment]
Thanks for reviewing, @astrofrog ! Your comments are addressed in the latest commit. And since I don't see any Sphinx test set up on Travis, I added `[skip ci]` to that commit (although it appears to run CircleCI anyway.

**Comment by astrofrog** [comment]
@pllim - just so you know, there is a Sphinx text (it's the one that says ``DOC_TRIGGER=1`` - not very obvious!

**Comment by pllim** [comment]
Ops. Just let me know if you want me to force push that last commit to re-trigger the entire CI suite.

**Comment by astrofrog** [comment]
Thanks @pllim!

**Pull request merged**